builtin/audit/socket/backend.go

```diff
@@ -177,6 +177,30 @@ func (b *Backend) LogResponse(ctx context.Context, in *logical.LogInput) error {
 	return err
 }
 
+func (b *Backend) LogTestMessage(ctx context.Context, in *logical.LogInput, config map[string]string) error {
+	var buf bytes.Buffer
+	temporaryFormatter := audit.NewTemporaryFormatter(config["format"], config["prefix"])
+	if err := temporaryFormatter.FormatRequest(ctx, &buf, b.formatConfig, in); err != nil {
+		return err
+	}
+
+	b.Lock()
+	defer b.Unlock()
+
+	err := b.write(ctx, buf.Bytes())
+	if err != nil {
+		rErr := b.reconnect(ctx)
+		if rErr != nil {
+			err = multierror.Append(err, rErr)
+		} else {
+			// Try once more after reconnecting
+			err = b.write(ctx, buf.Bytes())
```

Comment:
is it possible for the two write errors to be different? if so, maybe we also want to `multierror.Append()`?

Reply:
Good question. I think the last error is really the only one that really needs to be reported, so by preference I wouldn't use multierror at all.  (The existing code does and so we see "connection timed out" or similar errors repeated twice; I just copied the logic from the normal LogRequest/LogResponse functions.)

I think if we successfully reconnected then the original error is not interesting to report, while *maybe* a case could be made that if the write failed and the reconnect failed then they might usefully report different information.